src/cart/crt_iv.c

```diff
@@ -2333,9 +2329,12 @@ crt_ivsync_rpc_issue(struct crt_ivns_internal *ivns_internal, uint32_t class_id,
 	}
 
 	rc = crt_req_send(corpc_req, handle_response_cb, iv_sync_cb);
-	if (rc != 0)
-		D_ERROR("crt_req_send(): "DF_RC"\n", DP_RC(rc));
+	if (rc != 0) {
+		DL_ERROR(rc, "crt_req_send() failed, not calling decref");
 
+		/* Do not call decref on this rpc below if req_send failed */
+		corpc_req = NULL;
```

Comment:
[Question] Should we set `corpc_req` to `NULL` regardless of the return value of `crt_req_send`? At least the API comment says:

```
/**
 * Send an RPC request. In the case of sending failure, CRT internally destroy
 * the request \a req. In the case of succeed, the \a req will be internally
 * destroyed when the reply received. User needs not call crt_req_decref() to
 * destroy the request in either case.
```

@frostedcmos, could you help confirm what we should do here, please?

Reply:
In this instance it doesnt matter as a check for corpc_req will happen only if rc != 0 (line 2350), but in general yes, you are correct, once you've sent out an rpc via crt_req_send you should no longer manually decref it on any subsequent failure

Reply:
OK, I see.